config/karma/karmaConfig.js

```diff
@@ -48,214 +49,122 @@ const headlessConfig = karmaConfig => {
-    if (isDesktop(browserOrPlatform)) {
-      const versionSafe = ensureBrowserVersionExistsOnBrowserStack(
-        browserOrPlatform,
-        version
-      );
-      if (availableOnWindows(browserOrPlatform)) {
-        browserStackConfigurationObjects[
-          `bs_${browserOrPlatform}_${versionSafe}_windows`
-        ] = {
-          base: "BrowserStack",
-          browser: browserOrPlatform,
-          browser_version: versionSafe,
-          os: "WINDOWS",
-          os_version: "10"
-        };
-      }
-      if (availableOnOSX(browserOrPlatform)) {
-        browserStackConfigurationObjects[
-          `bs_${browserOrPlatform}_${versionSafe}_os_x`
-        ] = {
-          base: "BrowserStack",
-          browser: browserOrPlatform,
-          browser_version: versionSafe,
-          os: "OS X",
-          os_version: ensureOSXVersionIsCorrect(browserOrPlatform, versionSafe)
-        };
-      }
-    } else {
-      const isIOS = browserOrPlatform === "ios_saf";
-      const { os_version, device } = isIOS
-        ? getOSVersionAndDeviceForMobileSafariVersion(version)
-        : getOSVersionAndDeviceForMobileChromeVersion(version);
-      browserStackConfigurationObjects[
-        `bs_${browserOrPlatform}_${device}_${os_version}`
-      ] = {
-        base: "BrowserStack",
-        device,
-        real_mobile: true,
-        os: isIOS ? "ios" : "android",
-        os_version
-      };
-    }
-  });
-  return {
-    browsers: Object.keys(browserStackConfigurationObjects),
-    customLaunchers: browserStackConfigurationObjects
-  };
+const mapBrowsersListToBrowserStackLaunchers = async () => {
+  return await browserList_browserStack // nosonar
+    .default({
+      username: process.env.BROWSERSTACK_USERNAME,
+      accessKey: process.env.BROWSERSTACK_ACCESS_KEY,
+      browserslist: {
+        queries: [
+          "ie >= 11",
+          "last 1 Android versions",
+          "last 1 Chrome versions",
+          "last 1 Firefox versions",
+          "last 1 Safari versions",
+          "last 1 iOS versions",
+          "last 1 Edge versions",
+          "last 1 Opera versions",
+        ], // TODO: pull in the package.json version
+      },
+      formatForSelenium: false,
+    })
+    .then((res) => {
+      return formatForKarma(res, "BrowserStack");
+    });
 };
 
-const fullConfig = karmaConfig => {
+const fullConfig = async (karmaConfig) => {
   if (!isBrowserStackAvailable()) {
     console.log("BrowserStack not available");
     process.exit(0);
   }
-
-  const browserslist = require("browserslist");
+  const allBrowsers = await mapBrowsersListToBrowserStackLaunchers();
   const {
     browsers: bsBrowsers,
-    customLaunchers: customBSLaunchers
-  } = mapBrowsersListToBrowserStackLaunchers(browserslist());
+    customLaunchers: customBSLaunchers,
+  } = allBrowsers;
+
+  console.log(`Using BrowserStack browsers: ${bsBrowsers}`);
+  console.log(`Using BrowserStack custom browsers: ${customBSLaunchers}`);
```

Comment:
Elected to leave this in because it makes debugging browser-specific issues easier. Previously, you'd have to dif throw the BrowserStack web GUI to get this information.